### PR TITLE
Added README to BannerBase

### LINKED ISSUE #6682: Create README for BannerBase component
### Description
Currently there is no documentation for the `BannerBase` component. 

### Technical Details
- Create a `README.md` file detailing the component

### Acceptance Criteria
- A `README.md` exists in `app/component-library/components/Banners/Banner/foundation/BannerBase` with comprehensive component documentation

### PR DESCRIPTION
**Pull request by brianacnguyen**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/coding_guidelines/CODING_GUIDELINES.md)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

_Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions,_
- Added README to BannerBase

**Screenshots/Recordings**
Screenshot 2023-07-24 at 11 41 19 PM

_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

**Issue**

fixes #6682

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: app/component-library/components/Banners/Banner/foundation/BannerBase/README.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+# BannerBase
+
+BannerBase is a foundational component, which serves as a base for any inline notification.
```
Comment (georgewrmarshall) [inline app/component-library/components/Banners/Banner/foundation/BannerBase/README.md:3]: Can we align the description with the insight reports for figma and extension?

Screenshot 2023-07-25 at 10 11 00 AM

Screenshot 2023-07-25 at 10 09 29 AM

Screenshot 2023-07-25 at 10 10 13 AM

```suggestion
BannerBase serves as a base for all banner variants. It contains  standard props such as information and related actions.
```

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Looking good! Left some suggestions

**Review by georgewrmarshall — APPROVED** [review]
LGTM!

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**